src/path_cache.rs

```diff
@@ -1014,7 +1257,20 @@ impl<N: Neighborhood + Sync> PathCache<N> {
     }
 
     fn get_chunk(&self, point: Point) -> &Chunk {
-        get_chunk!(self, point)
+        let index = self.get_chunk_index(point);
+        get_chunk!(self, index)
+    }
+
+    fn get_chunk_index(&self, point: Point) -> usize {
+        let size = self.config.chunk_size;
+
+        let (x, y) = ((point.0 / size), (point.1 / size));
+        let chunk_count_x = match self.width % size {
+            0 => self.width / size,
+            _ => (self.width / size) + 1,
+        };
```

Comment:
The same check happens in `new_internal`, so just add a member to PathCache to store this